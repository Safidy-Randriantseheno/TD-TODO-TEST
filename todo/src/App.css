.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: wheat;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
  #form{
    background-color: rgba(0, 0, 0, 0.349);
    margin-left: 32%;
    padding: 5px;
    margin-top: 15px;
    text-align: center;
    width: 490px;
  }
  #todo{
    background-color: rgba(0, 0, 0, 0.349);
    width: 18%;
    height: 300px;
    border-radius: 8px;
  }
  #done{
    background-color: rgba(0, 0, 0, 0.349);
    width: 18%;
    height: 300px;
    margin-left: 5px;
    border-radius: 8px;
    
  }
  #container{
    margin-top: 20px;
  display: flex;
  flex-direction: row;
  justify-content: center;
  
   
  }
  .todo{
  background-color: white;
  width: 40%;
  justify-content: center;
  margin-left: 30%;
  border-radius: 8px;
  
  }
  .done{
    background-color: white;
    width: 40%;
    margin-left: 30%;
    border-radius: 8px;
    }
  

