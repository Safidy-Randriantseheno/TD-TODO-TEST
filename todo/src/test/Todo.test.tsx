import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import React from 'react';
import TodoApp from '../Todo';
import App from '../App';

test('typing into input and displaying todo', () => {
    render(<App />);
    const  inputElement = screen.getByPlaceholderText(/Enter new task/i)
    fireEvent.change(inputElement, { target: { value: 'safidy' } });
});

  it('renders correctly', () => {
    const  Placeholder  = render(<TodoApp />);
    const input = Placeholder.getByPlaceholderText('Enter new task');

    expect(input).toMatchSnapshot();
  });
  it('renders correctly', () => {
    const  Todo  = render(<TodoApp />);
    const todoHeading = Todo.getByText('TODO');

    expect(todoHeading).toMatchSnapshot();
  });
  it('renders correctly', () => {
    const  Done  = render(<TodoApp />);
    const doneHeading = Done.getByText('DONE');

    expect(doneHeading).toMatchSnapshot();
  });
  
  
  it('adding  new todo', () => {
    const { getByPlaceholderText, getByText } = render(<TodoApp />);
    const input = getByPlaceholderText('Enter new task');  

    fireEvent.change(input, { target: { value: 'Test Todo' } });
    fireEvent.submit(form);

    const newTodo = getByText('Test Todo');

    expect(newTodo).toBeInTheDocument();
  });
  it('todo completed', () => {
    const { getByPlaceholderText, getByText } = render(<TodoApp />);
    const input = getByPlaceholderText('Enter new task');
  
    const todoHeading = getByText('TODO');
    const doneHeading = getByText('DONE');

    fireEvent.change(input, { target: { value: 'Test Todo' } });
    fireEvent.submit(form);

    const newTodo = getByText('Test Todo');
    const checkbox = newTodo.previousSibling;

    fireEvent.click(checkbox);

    expect(newTodo).not.toBeInTheDocument();
    expect(doneHeading).toBeInTheDocument();
    expect(todoHeading).toBeInTheDocument();
  });
 