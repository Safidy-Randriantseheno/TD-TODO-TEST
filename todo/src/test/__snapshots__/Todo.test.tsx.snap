// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<input
  placeholder="Enter new task"
  type="text"
  value=""
/>
`;

exports[`renders correctly 2`] = `
<h2
  class="todo"
>
  TODO
</h2>
`;

exports[`renders correctly 3`] = `
<h2
  class="done"
>
  DONE
</h2>
`;
